### ISSUE TITLE
Domestic Trade Button disable

### ISSUE THREAD

**Issue by yeyyosing** [thyer/CS340winter2016#527]
After I make a trade, i am not allowed to attempt any more trades on my turn, since the domestic trade button is disabled.  To reproduce, on your turn, offer a trade, have the other person reject or accept the trade, and you will not be able to trade.

**Comment by eaglewyng** [comment]
This is valid. We're working on it, thanks!

**Comment by matthewfcarlson** [comment]
There's a funny story behind this one....

**Comment by matthewfcarlson** [comment]
I'll work on this @eaglewyng

**Comment by thyer** [comment]
This doesn't appear to be fixed

**Comment by sadler-j** [comment]
I had fixed this, but my fix caused other issues, so the team reverted the commit. I was informed of this after the submission. So you are correct, this is no longer fixed.

**Comment by thyer** [comment]
@sadler-j @eaglewyng et al. I'm discovering that the vast majority of these bugs are still not fixed in your latest release...what happened? Are any of you available to come talk to me so we can get it figured out?

**Comment by matthewfcarlson** [comment]
Yeah. Jonathan is on his way.

-Matthew Carlson

On Tue, Apr 5, 2016 at 10:21 AM, Trenton Adam Hyer <notifications@github.com

**Comment by eaglewyng** [comment]
Hey @thyer, I'm sorry I couldn't come talk to you today. When are you available tomorrow?

**Comment by thyer** [comment]
Jonathan and I talked for a bit today, I'm hoping we got enough details transmitted to get things fixed. Speaking of which, @sadler-j did you manage to dupe on the lab machines?

**Comment by matthewfcarlson** [comment]
Parker and I played around with it and I wasn't able to dupe it. I traded with other places and placed roads (which are two cases where it used to happen when the bug was filed).

**Comment by sadler-j** [comment]
I was going to try a lab machine, but it was during devotional so it wouldn't let me. I was going to try later and haven't had a chance yet. I was going to try running it normally and try running it like you were running it.

**Comment by eaglewyng** [comment]
Also, it seems as if some debugging code that was having the clients roll a 7 on every single turn was enabled. This debugging code wasn't in our official source at the time of commit, so we disabled it and re uploaded the jar on our 3.2 release. The player clients now shouldn't roll a 7 every time.

**Comment by thyer** [comment]
From what I showed Jonathan, there seemed to be something much different going on -- for example, I would start the setup phase, place a settlement, and then be prompted to place a road despite the placeSettlement command never making it to the server. It indefinitely froze the game. There were some workarounds, but the game kept breaking before I could ever get to thoroughly test most issues.

**Comment by sadler-j** [comment]
Agreed. And with the work arounds, I don't think there was an issue with rolling.

**Comment by sadler-j** [comment]
@thyer I just tested it in the sports lab. I tried it two different ways. The first was via separate terminals (which is the way we normally test it) and the other was using nohup (which is the way you have been testing it). It ran smoothly both ways. I didn't notice any differences between the runs. No bugs came up in the log either (at least on the server, I'm not sure how to check the client with nohup). This suggest the issue is either specific to the computer you are on or the version that is being run isn't 0.3.2. Did you ever download are 0.3 pre-release that could be run accidentally?

**Comment by thyer** [comment]
@sadler-j I'm going to try it on a new computer in the Pizza Lab.

**Comment by thyer** [comment]
@sadler-j Just got the same issues. I'm going to completely wipe everything, redownload your repo from https://github.com/BYUCS340/Catan/releases/tag/0.3.2 and start over

**Comment by sadler-j** [comment]
@thyer Let me know how that goes. If that doesn't work, I'm going to be incredibly befuddled.

**Comment by thyer** [comment]
Let the befuddling begin -- same issue, this time I may have discovered some more by accident. Let me give the step by step instructions for duplication with the new stuff in bold:

1. Run ant our-server in one terminal window
2. Run nohup ant our-client & in a second terminal window four times
3. In the first client window, register as "Trent1", create game "Trent's Game" and join as red. This should join you, but take you back to the Game Lobby page. **Do not rejoin the game and get to the PlayerWaiting modal!!**
4. Register in the remaining three clients as Trent2-Trent4. Join the game as orange, yellow, and white respectively.
5. The second you join the game in the fourth client, jump **immediately back** to the first client (where it should still be in the Game Lobby). **Notice that the setup modal appears and requests you to build a settlement, but when the poller fires, it brings you back to the lobby. If you build a settlement before the poller updates, it will properly register. If you do not build this settlement, it will walk you through the normal setup phase, but the settlement won't be built.**

**Comment by matthewfcarlson** [comment]
Ah yes. I think I have a good idea of what's going on. The client joins the game not the server after you create a new game. So the client's game is set to be the game you created even though you haven't selected re-join yet. When the game starts, the game launches and goes to an inconsistent state. That's my best guess.

**Comment by thyer** [comment]
@sadler-j can you verify? Can I provide more information?

**Comment by sadler-j** [comment]
I won't be able to test it until later. I can see the order of events potentially throwing things off (as described by @matthewfcarlson above). Have you tried running it by re-joining the game on the first client?

**Comment by thyer** [comment]
Yes, still some major crashing.

**Comment by sadler-j** [comment]
That's befuddling.

**Comment by matthewfcarlson** [comment]
So far working on my machine and the machines in the lab, if I rejoin the game on the first client it works reliably every time.
